src/script/components/MessagesList/index.tsx

```diff
@@ -182,14 +204,39 @@ const MessagesList: React.FC<MessagesListParams> = ({
   };
 
   // Listen to resizes of the the container element (if it's resized it means something has changed in the message list)
-  useResizeObserver(messagesContainer, () => updateScroll(messagesContainer));
+  useResizeObserver(() => updateScroll(messageContainerRef.current), messageContainerRef.current);
   // Also listen to the scrolling container resizes (when the window resizes or the inputBar changes)
-  useResizeObserver(messagesContainer?.parentElement, () => updateScroll(messagesContainer));
-  useLayoutEffect(() => {
-    if (messagesContainer) {
-      updateScroll(messagesContainer);
+  useResizeObserver(() => updateScroll(messageContainerRef.current), messageContainerRef.current?.parentElement);
+
+  const loadPrecedingMessages = async (): Promise<void> => {
+    const shouldPullMessages = !isPending && hasAdditionalMessages;
+
+    if (shouldPullMessages) {
+      await conversationRepository.getPrecedingMessages(conversation);
+    }
+  };
+
+  const loadFollowingMessages = () => {
+    const lastMessage = conversation.getLastMessage();
+
+    if (lastMessage) {
+      if (!isLastReceivedMessage(lastMessage, conversation)) {
+        // if the last loaded message is not the last of the conversation, we load the subsequent messages
+        if (isContentMessage(lastMessage)) {
+          conversationRepository.getSubsequentMessages(conversation, lastMessage);
+        }
+      }
+    }
+  };
+
+  useHitTopOrBottom(loadPrecedingMessages, loadFollowingMessages, messageListRef.current);
+  useFadingScrollbar(messageListRef.current);
```

Comment:
This should not be used this way. You should put it straight into the `ref` callback. 

```jsx
<div ref={element => {
  initFadingScrollbar(element); 
  hitToOfBottom(loadPreceding, loadFollowing, element)
}}></div>
```